utils.js

```diff
@@ -1842,10 +1835,97 @@ export async function parsePnpmLock(pnpmLock, parentComponent = null) {
           possibleOptionalDeps[decodeURIComponent(dpurl)] = false;
         }
       }
+
       dependenciesList.push({
         ref: decodeURIComponent(ppurl),
         dependsOn: ddeplist,
       });
+
+      // pnpm-lock.yaml contains more than root dependencies in importers
+      // TODO: verify if this is a good way to check.
+      // we do what we did above but for all the other components
+      for (const importedComponentName of Object.keys(yamlObj['importers'])) {
+
+        const componentDeps = yamlObj['importers'][importedComponentName]['dependencies'] || {};
+        const componentDevDeps = yamlObj['importers'][importedComponentName]['devDependencies'] || {};
+        const optionalDeps = yamlObj['importers'][importedComponentName]['optionalDependencies'] || {};
+        
+        let name = importedComponentName.split('/');
+        let lastname = name[name.length -1];
+
+        let subpath = name.filter(part => part !== '.' && part !== '..').join('/');
```

Comment:
Is there a better way to sanitize and remove relative directory names from the path.